traits/observers/expression.py

```diff
@@ -76,11 +85,82 @@ def then(self, expression):
         """
         return SeriesExpression(self, expression)
 
+    def dict_items(self, notify=True, optional=False):
+        """ Create a new expression for observing items inside a dict.
+
+        Events emitted (if any) will be instances of
+        :py:class:`~traits.observers.events.DictChangeEvent`.
+
+        If an expression with ``dict_items`` is further extended, the
+        **values** of the dict will be given to the next item in the
+        expression. For example, the following observes a trait named "number"
+        on any object that is one of the values in the dict named "mapping"::
+
+            trait("mapping").dict_items().trait("number")
+
+        Parameters
+        ----------
+        notify : boolean, optional
+            Whether to notify for changes.
+        optional : booleal, optional
+            Whether to ignore this if the upstream object is not a dict.
+
+        Returns
+        -------
+        new_expression : traits.observers.expressions.Expression
+        """
+        return self.then(dict_items(notify=notify, optional=optional))
+
+    def list_items(self, notify=True, optional=False):
+        """ Create a new expression for observing items inside a list.
+
+        Events emitted (if any) will be instances of
+        :py:class:`~traits.observers.events.ListChangeEvent`.
+
+        e.g. ``trait("containers").list_items()`` for observing to mutations
+        to a list named ``containers``.
+
+        e.g. ``trait("containers").list_items().trait("value")`` for observing
+        the trait ``value`` on any items in the list ``containers``.
+
+        Parameters
+        ----------
+        notify : boolean, optional
+            Whether to notify for changes.
+        optional : booleal, optional
```

Comment:
typo: `boolean`. (Incidentally, what's the rationale for using `boolean` rather `bool` here and elsewhere? `bool` is more standard and common. I know we're duck-typing, so technically we accept anything bool-like, but I'm not sure there's value in making that distinction at this level.)